File: Data Structures & Algorithms/duplicate-integer/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        const nums_set = new Set();
        
        for (let i = 0; i < nums.length; ++i) {
            if (nums_set.has(nums[i])) {
                return true;
            }
            nums_set.add(nums[i]);
        }
        return false;
    }
}

File: Data Structures & Algorithms/jump-game/submission-2.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        dp = [True] + [False for x in range(len(nums) - 1)]

        for i in range(len(nums)):
            if dp[i]:
                for j in range(nums[i] + 1):
                    if i + j < len(nums):
                        dp[i + j] = True
        
        return dp[len(nums) - 1]